core/config/value.go

```diff
@@ -487,8 +486,26 @@ func (cv Value) getValueStruct(key string, target interface{}) (interface{}, boo
 			if destSlice.Len() > 0 {
 				fieldValue.Set(destSlice)
 			}
-		}
+		case bucketMap:
+			createMap := reflect.MakeMap(fieldType).Interface()
+			val := global.GetValue(childKey).Value()
 
+			if val != nil {
+				switch v := val.(type) {
+				case map[interface{}]interface{}:
+					destMap := createMap.(map[string]interface{})
+					for key, value := range v {
+						switch key := key.(type) {
+						case string:
+							destMap[key] = value
+						default:
+							panic(fmt.Sprintf("Can't parse the input yaml. Map key must be string: %v", key))
```

Comment:
Yea, silently passing would be bad. Panicing doesn't seem like the right option. This is in `getValueStruct` which is called during `PopulateStruct`, which could be called in the middle of an app's lifecycle, not only during startup. Panic would be acceptable if it was only during app initialization, but we don't want UberFx to **ever** panic during a running (properly configured) application.

We do have the option to `return false` from PopulateStruct, which seems like the right thing to do here -- indicate that we couldn't deserialize properly. So we should probably log an error here, and then `return nil, false, errors.New("couldn't...")` (the middle value being the "ok") one so that the caller knows the PopulateStruct didn't work right.

Also we should follow up and have `PopulateStruct` return a proper `error` (or `nil`) rather than just a `bool`.